packages/openapi-ts/test/__snapshots__/test/generated/v3_angular_transform/types.gen.ts.snap

```diff
@@ -0,0 +1,132 @@
+// This file is auto-generated by @hey-api/openapi-ts
+
+/**
+ * This is a model that contains a some dates
+ */
+export type SimpleModel = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+};
+
+/**
+ * This is a model that contains a some dates
+ */
+export type ModelWithDates = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+    readonly modified: Date;
+    readonly expires?: Date;
+};
+
+export function ModelWithDates(data: any): ModelWithDates {
+    if (data?.modified) {
+        data.modified = new Date(data.modified);
+    }
+    if (data?.expires) {
+        data.expires = new Date(data.expires);
+    }
+    return data;
+}
+
+/**
+ * This is a model that contains a some dates and arrays
+ */
+export type ParentModelWithDates = {
+    id: number;
+    readonly modified?: Date;
+    items?: Array<ModelWithDates>;
+    item?: ModelWithDates;
+    simpleItems?: Array<SimpleModel>;
+    simpleItem?: SimpleModel;
+    dates?: Array<(Date)>;
+    strings?: Array<(string)>;
+};
+
+export function ParentModelWithDates(data: any): ParentModelWithDates {
```

Comment:
Think we will want to update these function names later, possibly move them out of types into services or their own file?

Reply:
The nice thing this way is they're just TS aliases which share the same name as the type (simplifies generation) and are co-located, but I'm broadly unopinionated